### DISCUSSION TITLE
Evaluating WAMR on Zephyr based nRF Connect SDK v1.5.1 and nRF52840-DK as HW platform

### DISCUSSION THREAD

**Discussion by irgu** [bytecodealliance/wasm-micro-runtime#658] [Q&A]
Hi, 

I would like to evaluate WAMR on this setup:
- Windows 10
- Zephyr based nRF Connect SDK v1.5.1 as SW platform
- SEGGER Embedded Studio as IDE
- nRF52840-DK as HW platform (Cortex-M4)

but it looks not easy currently. Maybe someone tested similiar setup or is able to point me whitch steps should be taken to achive that? I tried with basic instructions https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/build_wamr.md#zephyr but probably it's not so straightforward for not tested/supported MCU and also lacks section for Windows user.

**Comment by wenyongh** [comment]
@irgu Maybe you can ref to sample littlevgl and gui:
https://github.com/bytecodealliance/wasm-micro-runtime/tree/main/samples/littlevgl#test-on-zephyr
https://github.com/bytecodealliance/wasm-micro-runtime/tree/main/samples/gui#test-on-zephyr
Thanks.

**Comment by bsingh19** [comment]
Getting started with Zephyr (Setup toolchain, compilation, board bring-up, Hello World example)!!

Systems: MacBook
Board : nRF52833 DK (Optional)

https://youtu.be/4emoE-8hsqg

**Reply by xwang98** [reply]
Awesome! @bsingh5266

**Comment by ivan2angelov** [comment]
@irgu I managed to add WAMR as submodule to Zephyr using [nRF5340 board](https://www.nordicsemi.com/products/nrf5340). 
I just opened up [this discussion](https://github.com/bytecodealliance/wasm-micro-runtime/discussions/1337) with hope to get traction from @wenyongh or anybody from bytecode alliance. 
WAMR is great technology and I am honestly surprised how mature is the port. Great work bytecodealliance!
